content/tutorial/tutorial.md

```diff
@@ -513,63 +519,64 @@ class Board extends React.Component {
+#### जटिल विशेषताएँ सरल बनें {#complex-features-become-simple}
 
-Immutability makes complex features much easier to implement. Later in this tutorial, we will implement a "time travel" feature that allows us to review the tic-tac-toe game's history and "jump back" to previous moves. This functionality isn't specific to games -- an ability to undo and redo certain actions is a common requirement in applications. Avoiding direct data mutation lets us keep previous versions of the game's history intact, and reuse them later.
+अपरिवर्तनीयता जटिल सुविधाओं को लागू करने में बहुत आसान बनाती है। इस ट्यूटोरियल में बाद में, हम एक "टाइम ट्रैवल" सुविधा को लागू करेंगे, जो हमें टिक-टैक-टो गेम के इतिहास की समीक्षा करने और पिछली चालों के लिए "वापस कूदने" की अनुमति देगा। यह कार्यक्षमता खेलों के लिए विशिष्ट नहीं है - कुछ कार्यों को पूर्ववत और फिर से करने की क्षमता अनुप्रयोगों में एक सामान्य आवश्यकता है। प्रत्यक्ष डेटा उत्परिवर्तन से बचने से हम खेल के इतिहास के पिछले संस्करणों को बरकरार रख सकते हैं, और बाद में उनका पुन: उपयोग कर सकते हैं।
 
-#### Detecting Changes {#detecting-changes}
+#### परिवर्तन का पता लगाना {#detecting-changes}
 
-Detecting changes in mutable objects is difficult because they are modified directly. This detection requires the mutable object to be compared to previous copies of itself and the entire object tree to be traversed.
+उत्परिवर्तनीय वस्तुओं में परिवर्तन का पता लगाना मुश्किल है क्योंकि वे सीधे संशोधित होते हैं। इसको पता लगाने के लिए परिवर्तनशील वस्तु की आवश्यकता होती है जिसकी पिछली प्रतियों की तुलना की जाए और पूरे ऑब्जेक्ट ट्री को ट्रैवर्स किया जाए।
 
-Detecting changes in immutable objects is considerably easier. If the immutable object that is being referenced is different than the previous one, then the object has changed.
+अपरिवर्तनीय वस्तुओं में परिवर्तनों का पता लगाना काफी आसान है। यदि संदर्भित की जा रही अपरिवर्तनीय वस्तु पिछले एक से भिन्न है, तो वह वस्तु बदल गई है।
 
-#### Determining When to Re-Render in React {#determining-when-to-re-render-in-react}
+#### React में री-रेंडर कब करना है यह निर्धारित करना {#determining-when-to-re-render-in-react}
 
-The main benefit of immutability is that it helps you build _pure components_ in React. Immutable data can easily determine if changes have been made which helps to determine when a component requires re-rendering.
+अपरिवर्तनीयता का मुख्य लाभ यह है कि यह आपको React में _pure Components_ बनाने में मदद करता है। अपरिवर्तनीय डेटा आसानी से निर्धारित कर सकते हैं कि क्या परिवर्तन किए गए हैं जो यह निर्धारित करने में मदद करता है कि एक कौम्पोनॅन्ट को फिर से री-रेंडरिंग की आवश्यकता कब होती है।
 
-You can learn more about `shouldComponentUpdate()` and how you can build *pure components* by reading [Optimizing Performance](/docs/optimizing-performance.html#examples).
+आप `shouldComponentUpdate()` के बारे में और जान सकते हैं कि कैसे [ऑप्टिमिज़िंग परफॉरमेंस](/docs/optimizing-performance.html#examples) पढ़कर आप *शुद्ध कौम्पोनॅन्ट* का निर्माण कर सकते हैं।
```

Comment:
revisit the sentence, does not seem grammatically correct

Reply:
```
[ऑप्टिमिज़िंग परफॉरमेंस](/docs/optimizing-performance.html#examples) पढ़ के आप shouldComponentUpdate() के बारे में और जानकारी प्राप्त कर सकते हैं और *pure कौम्पोनॅन्ट* बनाने के बारे में जान सकते हैं
```